### PR TITLE
22246 Provide #digitSum in Integer + tests

### PR DESCRIPTION
**Pull request by astares**
https://pharo.fogbugz.com/f/cases/22246/Provide-digitSum-in-Integer-tests

### CODE REVIEW

**Review by SergeStinckwich — APPROVED** [review]
I agree with the changes, but do we need such a method ?

**Comment by astares** [comment]
We do not directly need it to run our image - but we have other methods that are also not
directly needed but good to have.

I would add it as 
 - it is useful for easy verification when number have to be entered (as an additional checksum)  
 - I often use it to demonstrate what one can easily do with Smalltalk 
   (it is a good example to understand #inject:into and one can apply it to very large numbers
    which is not possible in most other languages)

File: src/Kernel/Integer.class.st [hunk 1]
```diff
@@ -951,6 +951,13 @@ Integer >> digitSubtract: arg [
 	^ sum normalize
 ]
 
+{ #category : #accessing }
+Integer >> digitSum [
+    "Returns the digit sum of the receiver"
+    
+    ^self asString inject: 0 into: [:value :new| value + new digitValue]
+]
```
Comment (pavel-krivanek) [inline src/Kernel/Integer.class.st]: returns bad values for negative numbers (like 8 for -9)

**Comment by Ducasse** [comment]
It would be good to have

"1234 digitSum >>> 10"

**Pull request merged**